addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1266,6 +1270,86 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Build/remove parallax.
+     *
+     * @see this.selectClass for parameters
+     */
+    parallax: function (previewMode, widgetValue, params) {
+        const isParallax = widgetValue === 'true';
+        this.$target.toggleClass('parallax', isParallax);
+        if (isParallax) {
+            this._initParallax();
+        } else {
+            this._removeParallax();
+        }
+        this.isParallax = isParallax;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeVisibility: function () {
+        let $el = this.$target.find('> .s_parallax_bg');
+        $el = $el[0] ? $el : this.$target;
+        const hasBackgroundImage = $el.css('background-image') !== 'none';
+        if (!hasBackgroundImage) {
+            this._removeParallax();
+            this.$target.removeClass('oe_img_bg');
+        }
+        const showParallaxOptions = hasBackgroundImage && !this.$target.hasClass('o_background_video');
+        return this._super(...arguments) && showParallaxOptions;
+    },
+    /**
+     * @private
+     */
+    _initParallax: function () {
+        this._updateBackgroundOptions();
+        // FIXME big hack, this should be improved
+        if (!this.isParallax) {
+            const colorPicker = this._requestUserValueWidgets('bg_color_opt')[0];
+            colorPicker._previewColor = 'black-50';
+            colorPicker._value = 'black-50';
+            colorPicker.notifyValueChange(false);
+        }
+    },
+    /**
+     * @private
+     */
+    _removeParallax: function () {
+        const $bg = this.$target.find('> .s_parallax_bg');
+        if ($bg.length) {
+            const urlImage = $bg.css('background-image');
+            if (urlImage.length) {
+                this.$target.css('background-image', urlImage);
+                this.$target.addClass('oe_img_bg');
+            }
```

Comment:
Ideally I would like that to be automatic. Below you call `_updateBackgroundOptions` which will change the options target. Those options should be the one responsible to reapply their option on the new target. Actually here, you lose the background-position with your code.

Reply:
You mean modify the function "_onOptionUpdate" ? I'm not sure how to do it.

Reply:
No, when you update the options target, the `notify` method is called which calls the `setTarget` method. Does not need to be generic for now: change the `setTarget` method of both background and background position option so that they re-apply their config to the new target.